```
2023-01-30T23:34:03.6896137Z === RUN   TestRemoveContainerForceRemoveRunning
2023-01-30T23:34:18.7106643Z     remove_test.go:104: assertion failed: error is not nil: Error response from daemon: Could not kill running container 17a7340988337c1b6efd2c416860dc16657fadfe7f7fa342058e2b0f9ce03951, cannot remove - tried to kill container, but did not receive an exit event
2023-01-30T23:34:30.8304590Z     main_test.go:32: assertion failed: error is not nil: Error response from daemon: Could not kill running container 17a7340988337c1b6efd2c416860dc16657fadfe7f7fa342058e2b0f9ce03951, cannot remove - tried to kill container, but did not receive an exit event: failed to remove 17a7340988337c1b6efd2c416860dc16657fadfe7f7fa342058e2b0f9ce03951
2023-01-30T23:34:30.8329964Z --- FAIL: TestRemoveContainerForceRemoveRunning (27.14s)
```
```
2023-01-30T23:37:29.8414898Z time="2023-01-30T23:34:37.052836600Z" level=debug msg="event forwarded" ns=moby topic=/tasks/exit type=containerd.events.TaskExit
2023-01-30T23:37:29.8416747Z time="2023-01-30T23:34:37.052836600Z" level=info msg=Span duration="507.9µs" endTime="2023-01-30 23:34:37.0533445 +0000 GMT m=+32.843161201" event="&TaskExit{ContainerID:17a7340988337c1b6efd2c416860dc16657fadfe7f7fa342058e2b0f9ce03951,ID:17a7340988337c1b6efd2c416860dc16657fadfe7f7fa342058e2b0f9ce03951,Pid:10512,ExitStatus:1067,ExitedAt:2023-01-30 23:34:06.684802 +0000 GMT m=+2.474859801,XXX_unrecognized:[],}" name=publishEvent parentSpanID=2f1071eeb76b596d spanID=c0e4b35803551f10 startTime="2023-01-30 23:34:37.0528366 +0000 GMT m=+32.842653301" topic=/tasks/exit traceID=a5a26f90cb9aa39410c5f8896eb3b60d
```

Note the timestamps. The exit event is being sent after the daemon gives up waiting. Whatever the cause, it's not the changes in this PR.

--- a/daemon/monitor.go
+++ b/daemon/monitor.go
@@ -156,7 +156,7 @@ func (daemon *Daemon) ProcessEvent(id string, e libcontainerdtypes.EventType, ei
 
 		daemon.LogContainerEvent(c, "oom")
 	case libcontainerdtypes.EventExit:
-		if int(ei.Pid) == c.Pid {
+		if ei.ProcessID == ei.ContainerID {
 			return daemon.handleContainerExit(c, &ei)
 		}
 

--- a/libcontainerd/local/local_windows.go
+++ b/libcontainerd/local/local_windows.go
@@ -461,7 +461,7 @@ func (ctr *container) Start(_ context.Context, _ string, withStdin bool, attachS
 		}
 	}()
 	t := &task{process: process{
-		id:         libcontainerdtypes.InitProcessName,
+		id:         ctr.id,
 		ctr:        ctr,
 		hcsProcess: newProcess,
 		waitCh:     make(chan struct{}),
@@ -495,7 +495,7 @@ func (ctr *container) Start(_ context.Context, _ string, withStdin bool, attachS
 	ctr.client.eventQ.Append(ctr.id, func() {
 		ei := libcontainerdtypes.EventInfo{
 			ContainerID: ctr.id,
-			ProcessID:   libcontainerdtypes.InitProcessName,
+			ProcessID:   t.id,
 			Pid:         pid,
 		}
 		ctr.client.logger.WithFields(logrus.Fields{
@@ -793,7 +793,7 @@ func (t *task) Pause(_ context.Context) error {
 	t.ctr.client.eventQ.Append(t.ctr.id, func() {
 		err := t.ctr.client.backend.ProcessEvent(t.ctr.id, libcontainerdtypes.EventPaused, libcontainerdtypes.EventInfo{
 			ContainerID: t.ctr.id,
-			ProcessID:   libcontainerdtypes.InitProcessName,
+			ProcessID:   t.id,
 		})
 		t.ctr.client.logger.WithFields(logrus.Fields{
 			"container": t.ctr.id,
@@ -834,7 +834,7 @@ func (t *task) Resume(ctx context.Context) error {
 	t.ctr.client.eventQ.Append(t.ctr.id, func() {
 		err := t.ctr.client.backend.ProcessEvent(t.ctr.id, libcontainerdtypes.EventResumed, libcontainerdtypes.EventInfo{
 			ContainerID: t.ctr.id,
-			ProcessID:   libcontainerdtypes.InitProcessName,
+			ProcessID:   t.id,
 		})
 		t.ctr.client.logger.WithFields(logrus.Fields{
 			"container": t.ctr.id,

--- a/libcontainerd/remote/client.go
+++ b/libcontainerd/remote/client.go
@@ -213,9 +213,9 @@ func (c *container) Start(ctx context.Context, checkpointDir string, withStdin b
 
 	t, err = c.c8dCtr.NewTask(ctx,
 		func(id string) (cio.IO, error) {
-			fifos := newFIFOSet(bundle, libcontainerdtypes.InitProcessName, withStdin, spec.Process.Terminal)
+			fifos := newFIFOSet(bundle, id, withStdin, spec.Process.Terminal)
 
-			rio, err = c.createIO(fifos, libcontainerdtypes.InitProcessName, stdinCloseSync, attachStdio)
+			rio, err = c.createIO(fifos, stdinCloseSync, attachStdio)
 			return rio, err
 		},
 		taskOpts...,
@@ -278,7 +278,7 @@ func (t *task) Exec(ctx context.Context, processID string, spec *specs.Process, 
 	}()
 
 	p, err = t.Task.Exec(ctx, processID, spec, func(id string) (cio.IO, error) {
-		rio, err = t.ctr.createIO(fifos, processID, stdinCloseSync, attachStdio)
+		rio, err = t.ctr.createIO(fifos, stdinCloseSync, attachStdio)
 		return rio, err
 	})
 	if err != nil {
@@ -489,7 +489,7 @@ func (c *container) Task(ctx context.Context) (libcontainerdtypes.Task, error) {
 
 // createIO creates the io to be used by a process
 // This needs to get a pointer to interface as upon closure the process may not have yet been registered
-func (c *container) createIO(fifos *cio.FIFOSet, processID string, stdinCloseSync chan containerd.Process, attachStdio libcontainerdtypes.StdioCallback) (cio.IO, error) {
+func (c *container) createIO(fifos *cio.FIFOSet, stdinCloseSync chan containerd.Process, attachStdio libcontainerdtypes.StdioCallback) (cio.IO, error) {
 	var (
 		io  *cio.DirectIO
 		err error

--- a/libcontainerd/types/types.go
+++ b/libcontainerd/types/types.go
@@ -99,6 +99,3 @@ type Task interface {
 
 // StdioCallback is called to connect a container or process stdio.
 type StdioCallback func(io *cio.DirectIO) (cio.IO, error)
-
-// InitProcessName is the name given to the first process of a container
-const InitProcessName = "init"

